devenv.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+trap "trap - SIGTERM && kill -- -$$" SIGINT SIGTERM EXIT
+
+wait_for() {
+    echo "waiting for $1"
+
+    for _ in $(seq 10) ; do
+      if pg_isready -d postgres -h localhost -p "$1" -U postgres 1>/dev/null 2>&1; then
+        echo "connector up"
+        return 0
+      fi
+      echo -n "."
+      sleep 1
+    done
+    echo
+    echo "FAIL waiting for $1"
+    exit 1
+}
+
+cargo pgx start
+wait_for 28814
+psql -h localhost -p 28814 -d template1 -c 'CREATE EXTENSION IF NOT EXISTS timescaledb';
+createdb -h localhost -p 28814 "$(whoami)"
+# This allows entr to work correctly on docker for mac
+export ENTR_INOTIFY_WORKAROUND=true
+
+# Note: this is not a comprehensive list of source files
+SOURCE_FILES="src migration"
+find ${SOURCE_FILES} | entr -n cargo pgx install > ~/compile.log &
```

Comment:
So "it depends" because it kinda depends on what the expected behaviour is. Depending on the change that you make to the extension code, it may be necessary to drop and recreate the extension (because the SQL definitions are out-of-sync with the code). I'm not sure that we're in the position now to be able to automatically do that correctly. Although I suppose it wouldn't hurt to just kill all backends.

Reply:
To rephrase:

If you're running tests, or `make gendoc` against the database, then it shouldn't be necessary to restart backends/postgres. If you're connecting a long-running thing like Promscale to the database, then it would probably make sense to restart the backend, but, depending on the change that you make, it might not be sufficient to restart the backend. You might additionally need to `DROP EXTENSION promscale CASCADE;` and restart Promscale. Obviously this would cause data loss.

Reply:
For now I would suggest that we don't restart anything, and see where that causes trouble. When we have trouble, we can solve that problem in its own context.